tests/test_x509_ext.py

```diff
@@ -1147,6 +1147,43 @@ def test_ne(self):
         assert gn != object()
 
 
+class TestOtherName(object):
+    def test_invalid_args(self):
+        with pytest.raises(TypeError):
+            x509.OtherName(b"notanobjectidentifier", b"derdata")
+
+        with pytest.raises(TypeError):
+            x509.OtherName(x509.ObjectIdentifier("1.2.3.4"), "notderdata")
+
+    def test_repr(self):
+        gn = x509.OtherName(x509.ObjectIdentifier("1.2.3.4"), b"derdata")
+        assert repr(gn) == (
+            "<OtherName(type_id=<ObjectIdentifier(oid=1.2.3.4, "
+            "name=Unknown OID)>, value=b'derdata')>"
```

Comment:
Everywhere else we use the native bytes repr, and I think consistency with
that makes sense.

On Sun, Jul 5, 2015 at 5:57 PM, Joshua Tauberer <notifications@github.com>
wrote:

> In tests/test_x509_ext.py
> <https://github.com/pyca/cryptography/pull/2108#discussion_r33899414>:
>
> > @@ -1147,6 +1147,43 @@ def test_ne(self):
> >          assert gn != object()
> >
> >
> > +class TestOtherName(object):
> > +    def test_invalid_args(self):
> > +        with pytest.raises(TypeError):
> > +            x509.OtherName(b"notanobjectidentifier", b"derdata")
> > +
> > +        with pytest.raises(TypeError):
> > +            x509.OtherName(x509.ObjectIdentifier("1.2.3.4"), "notderdata")
> > +
> > +    def test_repr(self):
> > +        gn = x509.OtherName(x509.ObjectIdentifier("1.2.3.4"), b"derdata")
> > +        assert repr(gn) == (
> > +            "<OtherName(type_id=<ObjectIdentifier(oid=1.2.3.4, "
> > +            "name=Unknown OID)>, value=b'derdata')>"
>
> Is it worth modifying repr so it returns an identical string on Py2 and
> Py3?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2108/files#r33899414>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084